WhitehousePetitions/ViewController.swift

```diff
@@ -54,16 +55,19 @@ class ViewController: UITableViewController {
     }
     
     private func submit(_ answer: String) {
-        if answer.isEmpty {
-            petitions = allPetitions
-            tableView.performSelector(onMainThread: #selector(UITableView.reloadData), with: nil, waitUntilDone: false)
-            return
-        }
-        for item in allPetitions {
-            if item.body.contains(answer){
-                petitions.append(item)
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
```

Comment:
Not following why you are using a Dispatch Async here to a global queue and then later switching back to main thread for the tableView updates?